analysis/taint/testdata/annotations/main.go

```diff
@@ -0,0 +1,87 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"strings"
+)
+
+// Specific problem-level config options can be set in annotations
+//argot:config lim SetOptions(max-alarms=2,unsafe-max-depth=2)
+
+// bar
+// It is also a source for the problem hybridDef also defined in the config.json
+// The sink for that problem is fmt.Println
```

Comment:
I feel like it could get confusing if sources and sinks are defined in different places. Maybe it won't be an issue if we have descriptive error messages

Reply:
It might be confusing but I think it's necessary because you can't annotate third-party code, but you might want to annotate yours. 
I was thinking that we could add more information about "why was this identified as a sink/source/sanitizer/etc.." when doing the analysis, e.g. "this was a sink because it matched rule with "method": ..., "package": ...," or "matched an annotation ...".

Reply:
Also it would be nice if we could inform the user why something didn't match. Sometimes I'll run the analysis and it doesn't report anything because it couldn't find the entrypoints due to a typo in the config. So having Argot display something like the n "closest" matches to a config code-identifier would probably help the user debug the issue.